README.md

```diff
@@ -53,15 +53,20 @@ fixed by removing the offending lines (or the entire file) from
 
 ## Installing on POSIX systems
 
-Run the `install.py` script to install Diffuse.  The script accepts several
-options for customising the installation behaviour.  Use the `--help` option
-to obtain a full listing.
+Diffuse build system is meson.
 
-Diffuse can be removed by running the `install.py` script with the `--remove`
-option.
+To install diffuse locally:
 
-Specify the `--destdir=` and `--files-only` options for the `install.py`
-script when using it to build a platform specific package of Diffuse.
+    meson builddir
```

Comment:
How does this work for those that don't have meson?

Reply:
Meson seems to be available pretty much everywhere, are you in a case where you can't install it?

Reply:
I'm saying that I shouldn't have to install something to install this. I looked, and I don't have meson now. (I've never heard of it.)

Reply:
Well, you always had to install something anyway (python3, PyGObject, etc..), Meson is yet another dependency (but dev only this time).

I chose Meson because it is rather new (2013) and has become more and more used in most open source projects, to the point it has become the common replacement for the old autotools (e.g. most GNOME projects are using Meson or are transitioning to, as you can see in this long yet [not exhaustive list](https://mesonbuild.com/Users.html)).

Reply:
I updated the PR description explaining the move to Meson, I hope it's clearer ;)

Reply:
And also, I welcome your criticisms @joyously, it makes me think about the communication I might have missed here and there.

Reply:
> you always had to install something anyway... Meson is yet another dependency (but dev only this time).

I didn't. I am a user, not a dev, for this, and I used install.py to install the program so I could use it. You have removed install.py and rely on a user installing Meson, to install Diffuse?

Reply:
Actually I would rather rely on the distributions packaging systems. Most of the time, local installation is mainly for devs, regular users install their preferred applications through their preferred packaging systems.

Moreover, the fact that `install.py` was a custom script had two pitfalls:
* More maintenance for the devs
* A high possibility of having bugs

For instance, one very bad thing that the `install.py` was doing during installation is that it was directly replacing character strings in the destination files. For instance, it was searching for the string `'/usr/'` and replacing it by the `prefix` variable, this is a pretty bad pattern. Meson, on the other hand, proposes this feature already: you define your input file, your output file, and the template variables that you want to replace. In the code, the variables looks like this: `@PREFIX@`, which makes it clear that the file is a template and that these lines are to be replaced eventually.

Reply:
@joyously I was tinkering with Flatpak and indeed Meson simplifies the whole process. So once this PR will be approved, I'll be able to continue with the PR #64 that allows to create a Flatpak package which I should be able to push on Flathub. Once all that is done, installing Diffuse on your machine would be as easy as `flatpak install com.github.mightycreak.Diffuse` (and it will be easily searchable in GNOME Software).

Reply:
I guess that's good, but I don't have flatpak either. (I'm on Ubuntu Studio.) But if it shows up in `Software`, that will work.

Reply:
If you know Snap, Flatpak is more or less the same. IIRC, Ubuntu Studio is actually a classic Ubuntu but with pre-installed packages for content creators. So of you're interested in trying Flatpak, you can set it up here: https://flatpak.org/setup/

The advantages of Flatpak is that it makes it easy for a developer to distribute its program to several distributions at once. And for the user, Flatpak brings more security by sandboxing each applications and warn the user of the different permissions needed by the application.

Reply:
FWIW meson is at least a very light dependency since it needs nothing but python3 (and py3-setuptools for installation) installed and it makes it significantly easier for distros (and others) to package a project due to its easy of use (and generally custom build scripts are prone to breaking when used on systems that differ from the dev's system). I should look into packaging this for Alpine Linux once meson support is merged :)

Reply:
Thanks @Cogitri!

@joyously I am sure this is better for the future of Diffuse, so I'll go and merge this PR. Thanks for the feedbacks, it allows me to know how Diffuse is actually used :wink: